pkg/services/ngalert/state/state.go

```diff
@@ -82,11 +82,62 @@ func (a *State) GetAlertInstanceKey() (models.AlertInstanceKey, error) {
 	return models.AlertInstanceKey{RuleOrgID: a.OrgID, RuleUID: a.AlertRuleUID, LabelsHash: labelsHash}, nil
 }
 
+// Alerting sets the state to Alerting. It changes both the start and end time.
+func (a *State) Alerting(startsAt, endsAt time.Time) {
```

Comment:
I think the better name would be SetAlerting(... as it reflects the actual mutation

Reply:
Done 🙂